android-sdk/src/main/java/webtrekk/android/sdk/extension/Extension.kt

```diff
@@ -50,3 +53,20 @@ fun <T : Any> List<T>?.validateEntireList(propertyName: Any): List<T> {
 }
 
 fun String.encodeToUTF8(): String = URLEncoder.encode(this, "UTF-8")
+fun String?.jsonToMap(): Map<String, String> {
+    val map = HashMap<String, String>()
+    try {
+        val obj = JSONObject(this)
+        val keysItr = obj.keys()
+        while (keysItr.hasNext()) {
+            val key = keysItr.next()
+            val value = obj.getString(key)
+            map[key] = value
+        }
+    } catch (e: JSONException) {
+        error(e)
```

Comment:
`error` will just throw `IllegalStateException`, and the exception is not caught in the high level when using this function which will lead to app crash. Just probably wrap it in `try` `catch` whenever you use this function.